### ISSUE TITLE
The current month cannot be reselected using the keyboard navigation.

### ISSUE THREAD

**Issue by simonssspirit** [telerik/kendo-ui-core#2776]
### Bug report

The current month cannot be reselected using the keyboard navigation. The issue occurs when the min property is set to current date, and the start is set to "year".

### Reproduction of the problem

The issue can be reproduced in the following Dojo example: 

Press the tab key until the datepicker control is active. Then use the keyboard shortcut 'alt+down cursor' to open the picker, then use the cursors to select a month. However, notice that if you move away from the initial month then the cursor keys will not allow reselecting the first available month.

The issue is reported for the MVC wrappers, but it was reproduced in the jQuery version as well.

### Environment

* **Browser:** [all]

**Comment by dbalinov** [comment]
http://dojo.telerik.com/OZahUy

**Comment by Xizario** [comment]
The issue still happens when `min` date is last day of the month (`31`/5/2017), and the next navigated month have only `30` days.
- http://dojo.telerik.com/OZahUy/9
- Open the calendar, press down then up. May can not be selected.
